### PR TITLE
SOLID Principles

### LINKED ISSUE #8: Borja - Module 5
Changes to be applied :

- Check inheritance vs composition.
- Singleton pattern.
- Single responsibility principle: A class should have one reason to change, or one responsibility.
- Open-closed principle: Objects should be open for extension but closed for modification
- Liskov substitution principle: A principle of OOP. 
- Interface segregation principle: A principle of OOP.
- Dependency inversion principle: A principle of OOP.

* Recommended a commit for each point

### PR DESCRIPTION
**Pull request by BorjaDelgadoDev**
**Single Responsibility Principle**
Improvements in the structure and organization of code into multiple classes. Applying this principle

**Open/Close principle**
Refactor classes for improved extensibility

**Interface segregation principle**
Refactoring on navigation interfaces

**Dependency inversion principle**
Refactored code to depend on abstractions for improved flexibility and maintainability

### CODE REVIEW

**Review by diego-matesanz — APPROVED** [review]
Approved!

File: app/src/main/java/com/group3/architectcoders/data/IBooksRepository.kt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.group3.architectcoders.data
+
+interface IBooksRepository {
+    suspend fun fetchBooksBySearchText(search: String): List<Book>
+    suspend fun fetchBookById(id: String): Book
+    suspend fun fetchBookByIsbn(isbn: String): Book
+}
```
Comment (cristiangoncas) [inline app/src/main/java/com/group3/architectcoders/data/IBooksRepository.kt:7]: Missing empty line at the end of the file

File: app/src/main/java/com/group3/architectcoders/ui/screens/navigation/BookNavigationAction.kt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package com.group3.architectcoders.ui.screens.navigation
+
+interface BookNavigationAction {
+    fun onBookClick(bookId: String)
+}
```
Comment (cristiangoncas) [inline app/src/main/java/com/group3/architectcoders/ui/screens/navigation/BookNavigationAction.kt:5]: Missing empty line at the end of the file

**Review by cristiangoncas — APPROVED** [review]
Looks good, just a couple of files missing an empty line at the end of the file.

**Review by Caticatt — APPROVED** [review]

**Pull request merged**